# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStore.cpp
# Commit: 56cce39fb13c
# Full Hash: 56cce39fb13cd0623644bf9d40cbcec3473a93d1
# Author: edgul <edgul@mozilla.com>
# Date: 2025-06-03 21:32:22
# Description:
#   Bug 1968495 - Pass NotNull nsURIs across cookiestore IPC. r=baku,cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251177
# ==============================================================================

diff -r e93a3c226679 -r 56cce39fb13c dom/cookiestore/CookieStore.cpp
--- a/dom/cookiestore/CookieStore.cpp	Tue Jun 03 13:21:06 2025 +0000
+++ b/dom/cookiestore/CookieStore.cpp	Tue Jun 03 13:21:35 2025 +0000
@@ -442,10 +442,10 @@
         nsCOMPtr<nsIURI> cookieURI = cookiePrincipal->GetURI();
         RefPtr<CookieStoreChild::SetRequestPromise> ipcPromise =
             self->mActor->SendSetRequest(
-                cookieURI, cookiePrincipal->OriginAttributesRef(),
-                thirdPartyContext, partitionForeign, usingStorageAccess,
-                isOn3PCBExceptionList, nsString(aOptions.mName),
-                nsString(aOptions.mValue),
+                mozilla::WrapNotNull(cookieURI.get()),
+                cookiePrincipal->OriginAttributesRef(), thirdPartyContext,
+                partitionForeign, usingStorageAccess, isOn3PCBExceptionList,
+                nsString(aOptions.mName), nsString(aOptions.mValue),
                 // If expires is not set, it's a session cookie.
                 aOptions.mExpires.IsNull(), ComputeExpiry(aOptions),
                 aOptions.mDomain, path, SameSiteToConst(aOptions.mSameSite),
@@ -555,11 +555,11 @@
         nsCOMPtr<nsIURI> cookieURI = cookiePrincipal->GetURI();
         RefPtr<CookieStoreChild::DeleteRequestPromise> ipcPromise =
             self->mActor->SendDeleteRequest(
-                cookieURI, cookiePrincipal->OriginAttributesRef(),
-                thirdPartyContext, partitionForeign, usingStorageAccess,
-                isOn3PCBExceptionList, nsString(aOptions.mName),
-                nsString(aOptions.mDomain), path, aOptions.mPartitioned,
-                operationID);
+                mozilla::WrapNotNull(cookieURI.get()),
+                cookiePrincipal->OriginAttributesRef(), thirdPartyContext,
+                partitionForeign, usingStorageAccess, isOn3PCBExceptionList,
+                nsString(aOptions.mName), nsString(aOptions.mDomain), path,
+                aOptions.mPartitioned, operationID);
         if (NS_WARN_IF(!ipcPromise)) {
           promise->MaybeResolveWithUndefined();
           return;
@@ -756,7 +756,8 @@
         nsCOMPtr<nsIURI> cookieURI = cookiePrincipal->GetURI();
         RefPtr<CookieStoreChild::GetRequestPromise> ipcPromise =
             self->mActor->SendGetRequest(
-                cookieURI, cookiePrincipal->OriginAttributesRef(),
+                mozilla::WrapNotNull(cookieURI.get()),
+                cookiePrincipal->OriginAttributesRef(),
                 partitionedCookiePrincipal
                     ? Some(partitionedCookiePrincipal->OriginAttributesRef())
                     : Nothing(),